test/integration/crud/explain.test.ts

```diff
@@ -0,0 +1,282 @@
+    const collection = db.collection('test');
+
+    await collection.insertOne({ a: 1 });
+    const aggResult = await collection
+      .aggregate([{ $project: { a: 1 } }, { $group: { _id: '$a' } }])
+      .explain(false);
+    if (aggResult && aggResult.stages) {
+      expect(aggResult.stages).to.have.length.gte(1);
+      expect(aggResult.stages[0]).to.have.property('$cursor');
+      expect(aggResult.stages[0].$cursor).to.have.property('queryPlanner');
+      expect(aggResult.stages[0].$cursor).to.not.have.property('executionStats');
+    } else if (aggResult.$cursor) {
+      expect(aggResult.$cursor).to.have.property('queryPlanner');
+      expect(aggResult.$cursor).to.not.have.property('executionStats');
+    } else {
+      expect(aggResult).to.have.property('queryPlanner');
+      expect(aggResult).to.not.have.property('executionStats');
+    }
+  });
+
+  it('should honor string explain specified on cursor with aggregate', async function () {
+    const db = client.db('shouldHonorStringExplainSpecifiedOnCursor');
+    const collection = db.collection('test');
+
+    await collection.insertOne({ a: 1 });
+    const aggResult = await collection
+      .aggregate([{ $project: { a: 1 } }, { $group: { _id: '$a' } }])
+      .explain('allPlansExecution');
+
+    if (aggResult && aggResult.stages) {
+      expect(aggResult.stages).to.have.length.gte(1);
+      expect(aggResult.stages[0]).to.have.property('$cursor');
+      expect(aggResult.stages[0].$cursor).to.have.property('queryPlanner');
+      expect(aggResult.stages[0].$cursor).to.have.property('executionStats');
+    } else {
+      expect(aggResult).to.have.property('queryPlanner');
+      expect(aggResult).to.have.property('executionStats');
+    }
+  });
+
+  it('should honor legacy explain with aggregate', async function () {
+    const db = client.db('shouldHonorLegacyExplainWithAggregate');
+    const collection = db.collection('test');
+
+    await collection.insertOne({ a: 1 });
+    const aggResult = await collection
+      .aggregate([{ $project: { a: 1 } }, { $group: { _id: '$a' } }])
+      .explain();
+    if (aggResult && aggResult.stages) {
+      expect(aggResult.stages).to.have.length.gte(1);
+      expect(aggResult.stages[0]).to.have.property('$cursor');
+      expect(aggResult.stages[0].$cursor).to.have.property('queryPlanner');
+      expect(aggResult.stages[0].$cursor).to.have.property('executionStats');
+    } else {
+      expect(aggResult).to.have.property('queryPlanner');
+      expect(aggResult).to.have.property('executionStats');
+    }
+  });
+
+  it('should throw a catchable error with invalid explain string', async function () {
+    const db = client.db('shouldThrowCatchableError');
+    const collection = db.collection('test');
+    const error = await collection
+      .find({ a: 1 })
+      .explain('invalidExplain')
+      .catch(error => error);
+    expect(error).to.be.instanceOf(MongoServerError);
+  });
```

Comment:
I would put this under a context block with a title like `when explain is set to an invalid value' and change the title of this test to follow suit.